requirements.txt

```diff
@@ -2,17 +2,17 @@ asn1crypto==1.5.1
 certifi==2023.7.22
 cffi==1.15.1
 charset-normalizer==2.1.1
-cryptography==41.0.6
+cryptography==42.0.5
 filelock==3.10.0
 idna==3.4
 oscrypto==1.3.0
 packaging==23.0
 pycparser==2.21
 pycryptodomex==3.17
 PyJWT==2.6.0
-pyOpenSSL==23.3.0
+pyOpenSSL==24.1.0
 pytz==2022.7.1
 requests==2.31.0    # Updated to match the version on platform and freeze it at this version.
```

Comment:
Can we take the advantage of the improvement made by @sodle-splunk https://github.com/phantomcyber/dev-cicd-tools/pull/7?

Reply:
isort won't help here much, but I have added it 👍🏻

Reply:
Updated the comment with correct link

Reply:
Upgraded tools to v1.18 - thank you!

Reply:
I was also hoping you would remove those pesky packages from requirements.txt :P

Reply:
@sodle-splunk This is another thing to keep in mind - some of those might be in requirements, but would then not be put in the wheels directory, avoiding that would help us avoid confusion and make this whole thing cleaner

Reply:
This might be an enhancement we pursue when converting this logic from pre-commit to GitHub Actions. 

I'm hesitant to automatically remove things from `requirements.txt`, as I believe it is used by developers to install dependencies locally when working on a connector. However, we could automatically add a GitHub comment suggesting that the dependency be removed or moved into `requirements-dev.txt`.

Reply:
Automatically is probably not the best idea, but without SOAR the connectors have rather slim chances of working anyways, right? I think at least checking for those is a good idea

Reply:
IDK if I should remove it here... that way while developing, the env won't see them and may be harder to set up for the engineer. 

But using requirements-dev.txt or sth similar could do the trick... 🤔 What do you think?

Reply:
Other PR has already removed these pesky packages from requirements.txt so I don't think this thread is still relevant to this PR and I will resolve it. 

However, the topic of how to handle the dev requirements is definitely worth discussing. I'm personally a fan of the idea of creating some kind of Connectors SDK, which will provide all the packages available in the platform. I think this topic can be researched and discussed in some ERD.